modules/dists/SparseBlockDist.chpl

```diff
@@ -117,6 +118,64 @@ class SparseBlockDom: BaseSparseDom {
     dsiAdd((ind,));
   }
 
+  proc dsiBulkAdd(inds: [] index(rank, idxType), isSorted=false, isUnique=false,
+      preserveInds=true){
+  
+    if !isSorted && preserveInds {
+      var _inds = inds;
+      bulkAdd_help(_inds, isSorted, isUnique); 
+    }
+    else {
+      bulkAdd_help(inds, isSorted, isUnique);
+    }
+  }
+
+  proc bulkAdd_help(inds: [] index(rank,idxType), isSorted=false, isUnique=false) {
+
+    if !isSorted then
+      QuickSort(inds);
```

Comment:
Also for sorting here: I think regardless of how the rest of the function is implemented, having inds sorted at this point would have the advantage of increased temporal cache locality while assigning indices to different `localIndexes` arrays. Otherwise, if the input index array is unsorted, we would be jumping around different arrays and counts.

Also, there shouldn't be additional overhead of sorting since bulkAdd is called with `isSorted=true`.

Does it make sense?